multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/resolvers/v2/ResourceLiveParameterResolver.java

```diff
@@ -0,0 +1,149 @@
+                                                                                         .equals(ref.getDependencyName()))
+                                                                         .findAny();
+
+        if (moduleProvidedDependencyOpt.isPresent()) {
+            Object requiredDependencyValue = moduleProvidedDependencyOpt.get()
+                                                                        .getProperties()
+                                                                        .get(ref.getKey());
+
+            if (requiredDependencyValue instanceof Map) {
+                resolveModuleProvidedDependency(castToMap(requiredDependencyValue), module);
+                resourceParameter.setValue(requiredDependencyValue);
+            } else if (isStringValueReplaceable(requiredDependencyValue)) {
+                replaceValueInResourceParameter(module, resourceParameterValue, resourceParameter, ref,
+                                                castToString(requiredDependencyValue));
+            }
+        }
+    }
+
+    private boolean isStringValueReplaceable(Object object) {
+        return object instanceof String && idleToLiveParameterPairs.keySet()
+                                                                   .stream()
+                                                                   .anyMatch(defaultParamPlaceholder -> castToString(object).contains(defaultParamPlaceholder));
+    }
+
+    private void replaceValueInResourceParameter(Module module, String resourceParameterValue, Map.Entry<String, Object> resourceParameter,
+                                                 Reference reference, String requiredDependencyValueString) {
+        Map.Entry<String, String> requiredIdleToLiveParameterPair = getMapEntryOnCondition(requiredDependencyValueString);
+        requiredDependencyValueString = requiredDependencyValueString.replace(requiredIdleToLiveParameterPair.getKey(),
+                                                                              castToString(module.getParameters()
+                                                                                                 .get(requiredIdleToLiveParameterPair.getValue())));
+        requiredDependencyValueString = resourceParameterValue.replace(reference.getMatchedPattern(), requiredDependencyValueString);
+
+        resourceParameter.setValue(requiredDependencyValueString);
+    }
+
+    private void resolveModuleProvidedDependency(Map<String, Object> requiredDependencyMap, Module module) {
+        requiredDependencyMap.entrySet()
+                             .forEach(entry -> updateModuleDependencyFromMap(module, entry));
+    }
+
+    private void updateModuleDependencyFromMap(Module module, Map.Entry<String, Object> entry) {
+        if (isStringValueReplaceable(entry.getValue())) {
+            var idleToLiveParameterPair = getMapEntryOnCondition(castToString(entry.getValue()));
```

Comment:
rename the variable and maybe the method to something more descriptive, e.g. 'var matchedPlaceholderEntry = matchPlaceholderInParameter(...)